### PR TITLE
Fix/62

### PR DESCRIPTION
**Pull request by bspitzbart**
This fixes compatibility with pytorch 1.3.  
Bento, are there any other changes? There are more differences than I was expecting.  Any new dependencies?   
Giannis, please check the imports. I think we need ..utils as before instead of just utils due the directory structure.

### CODE REVIEW

**Review by iparask — CHANGES_REQUESTED** [review]
Hello @bspitzbart ,

The execution fails with a syntax error:
```
(SealsPytorch1) [paraskev@gpu045 unit.000066]$ iceberg_seals.predicting "--input_image" "WV03_20141120084907_1040010004AD7600_14NOV20084907-P1BS-500268558060_01_P001_u08rf3031.tif" "--model_architecture" "UnetCntWRN" "--hyperparameter_set" "A" "--training_set" "test_vanilla" "--test_folder" "/pylon5/mc3bggp/paraskev/SealsTest/test_image" "--model_path" "./" "--output_folder" "./WV03_20141120084907_1040010004AD7600_14NOV20084907-P1BS-500268558060_01_P001_u08rf3031"
Traceback (most recent call last):
  File "/pylon5/mc3bggp/paraskev/SealsPytorch1/bin/iceberg_seals.predicting", line 20, in <module>
    from iceberg_seals.search.predicting.predict_sealnet import predict_patch
  File "/pylon5/mc3bggp/paraskev/SealsPytorch1/lib/python3.5/site-packages/iceberg_seals/search/predicting/predict_sealnet.py", line 114
    os.makedirs(f'{output_dir}/heatmaps', exist_ok=True)
                                       ^
SyntaxError: invalid syntax
(SealsPytorch1) [paraskev@gpu045 unit.000066]$
```

In addition, may I suggest that the setup file to be changed and remove the version dependencies to torch, torchvision, and tensorboardX?

**Comment by bspitzbart** [comment]
@bentocg Can you weigh in on the syntax error?  Also can we eliminate all version dependencies or do some require specific versions?  Thanks.

**Comment by iparask** [comment]
Hello @bspitzbart, I fixed all the dependencies, syntax errors and the predicting binary. Please have a look and let me know if you have any questions. I would like to release the fix today if possible.

File: src/iceberg_seals/search/predicting/predict_sealnet.py [hunk 1]
```diff
@@ -45,8 +46,8 @@ def parse_args():
                                                        'used in subsequent steps of the pipeline')
     parser.add_argument('--models_folder', type=str, default='saved_models', help='folder where the model tar file is'
                                                                                   'saved')
-    parser.add_argument('--input_image', type=str, help='filename of input raster')
     parser.add_argument('--output_dir', type=str, help='folder where output shapefiles will be stored')
+    parser.add_argument('--save_heatmaps', type=int, help='boolean for saving heatmaps', default=0)
```
Comment (iparask) [inline src/iceberg_seals/search/predicting/predict_sealnet.py:50]: @bspitzbart this value is an int, while the help message says boolean. In addition, it is passed to a boolean parameter later. I would suggest to change it to Boolean and set as the default `True`

File: src/iceberg_seals/search/predicting/predict_sealnet.py [hunk 2]
```diff
@@ -274,8 +292,7 @@ def main():
                                                                 args.model_name)))
 
     # run validation to get confusion matrix
-    predict_patch(input_image=args.input_image, model=model,
-                  input_size=model_archs[args.model_architecture]['input_size'],
+    predict_patch(model=model, input_size=model_archs[args.model_architecture]['input_size'],
```
Comment (iparask) [inline src/iceberg_seals/search/predicting/predict_sealnet.py:295]: `save_heatmaps` is not passed here, although it is given as an input. It should be added.

**Review by iparask — CHANGES_REQUESTED** [review]

**Comment by iparask** [comment]
This is not relevant any more

**Pull request closed without merging**